src/plugins/intel_gpu/src/graph/fully_connected.cpp

```diff
@@ -105,11 +107,43 @@ layout fully_connected_inst::calc_output_layout(fully_connected_node const& node
     if (desc->input_size == 3) {
         output_size = tensor(input_layout.batch(), input_layout.feature(), 1, weights_layout.batch());
     }
-    format output_format = get_preferred_format(node, impl_param);
+    format output_format = get_preferred_format(impl_param);
 
     return layout(output_type, output_format, output_size);
 }
 
+template<typename ShapeType>
+std::vector<layout> fully_connected_inst::calc_output_layouts(fully_connected_node const& /*node*/, const kernel_impl_params& impl_param) {
+    auto desc = impl_param.typed_desc<fully_connected>();
+    auto input_layout = impl_param.get_input_layout();
+    auto weights_layout = *impl_param.weights_layout;
+
+    auto output_type = input_layout.data_type;
+    if ((output_type == data_types::u8 || output_type == data_types::i8) && desc->output_data_type)
```

Comment:
This condition looks a bit weird... I wonder how it works in unit tests when we have u8/i8 inputs, but don't set output_data_type... seems like it puts u8/i8 output data type which seems to be invalid...
I think it should be something like this:

```
    auto default_out_dt = data_type_traits::is_floating_point(input_layout.data_type) ? input_layout.data_type : data_types::f32;
    auto output_type = desc->output_data_type.value_or(default_out_dt);

    if (impl_param.has_fused_primitives()) {
        output_type = impl_param.get_fused_output_layout().data_type;
    }

```